client/analytics/report/revenue/index.js

```diff
@@ -70,11 +72,20 @@ class RevenueReport extends Component {
 
 			response.json().then( () => {
 				// Ignore data, just use our fake data once we have a response
-				this.setState( { stats: rawData } );
+				this.setState( { stats: mockData } );
 			} );
 		} );
 	}
 
+	onDownload( headers, rows, totals, query ) {
+		return () => {
```

Comment:
Should we add a `TODO` note here about implementing a "full" download solution if the data set is larger than what is contained in the client side data?

Reply:
Also since these methods will likely be used on all reports, what do you think might be a good way to abstract this?

Reply:
> Also since these methods will likely be used on all reports, what do you think might be a good way to abstract this?

While coding that, I wondered if a `downloadCSVFileFromTable()` function would make sense, but didn't implement it because I was unsure.

Indeed, at some point, it might be better to create a React Component named `<TableReportDownloadButton>` (or something like that) which implements the drop-down and calls the CSV download methods as well as handling long reports, sending reports by email, etc. That component would replace the current 'Download' button in tables.